### PR TITLE
PTX-15067:Torpedo job doesn't fail if core was found on the node

### PR DESCRIPTION
**Pull request by CharudathGopal**
- Made change to fail the test upon core found

### CODE REVIEW

File: tests/basic/ipv6_test.go [hunk 1]
```diff
@@ -26,7 +26,9 @@ var _ = Describe("{IPv6PxctlFunctional}", func() {
 		runID = testrailuttils.AddRunsToMilestone(testrailID)
 		nodes = node.GetWorkerNodes()
 		numNodes = len(nodes)
-		StartTorpedoTest("IPv6PxctlCommands", "Pxctl commands on IPv6 setups test", nil, testrailID)
+		tags := make(map[string]string, 0)
+		tags["ipv6"] = "true"
```
Comment (mborodin-px) [inline tests/basic/ipv6_test.go:30]: you can initialize map[string]string with values like this:
```
tags := map[string]string{
		"ipv6": true,
}
```

File: tests/basic/longevity_test.go [hunk 2]
```diff
@@ -108,7 +108,9 @@ var _ = Describe("{Longevity}", func() {
 
 	BeforeEach(func() {
 		if !populateDone {
-			StartTorpedoTest("PX-Longevity", "Validate PX longevity workflow", nil, 0)
+			tags := make(map[string]string, 0)
+			tags["longevity"] = "true"
```
Comment (mborodin-px) [inline tests/basic/longevity_test.go:112]: the same as above

File: tests/basic/unhappypath_test.go [hunk 3]
```diff
@@ -41,7 +41,9 @@ var _ = Describe("{NetworkErrorInjection}", func() {
 	// testrailID corresponds to: https://portworx.testrail.net/index.php?/cases/view/35264
 	var runID int
 	JustBeforeEach(func() {
-		StartTorpedoTest("NetworkErrorInjection", "Verify stability of the system when there is a network error", nil, testrailID)
+		tags := make(map[string]string, 0)
+		tags["networkErrorInjection"] = "true"
```
Comment (mborodin-px) [inline tests/basic/unhappypath_test.go:45]: the same as above

File: tests/basic/upgrade_cluster_test.go [hunk 4]
```diff
@@ -15,7 +15,9 @@ var _ = Describe("{UpgradeCluster}", func() {
 	var contexts []*scheduler.Context
 
 	JustBeforeEach(func() {
-		StartTorpedoTest("UpgradeCluster", "Upgrade cluster test", nil, 0)
+		tags := make(map[string]string, 0)
+		tags["upgradeCluster"] = "true"
```
Comment (mborodin-px) [inline tests/basic/upgrade_cluster_test.go:19]: the same as above

File: tests/common.go [hunk 5]
```diff
@@ -1827,12 +1828,18 @@ func PerformSystemCheck() {
 					TimeBeforeRetry: 10 * time.Second,
 				})
 				if len(file) != 0 || err != nil {
-					log.Errorf("an error occurred, collecting bundle")
-					CollectSupport()
-					dash.VerifySafely(file == "", true, fmt.Sprintf("Core generated on node %s, Core Path: %s", n.Name, file))
+					dash.Errorf("Core generated on node %s, Core Path: %s", n.Name, file)
```
Comment (mborodin-px) [inline tests/common.go:1831]: I would re-phrase the error message to `Core file was found on node %n`

File: tests/common.go [hunk 6]
```diff
@@ -1827,12 +1828,18 @@ func PerformSystemCheck() {
...
+					// Collect Support Bundle only once
+					if !foundCore {
+						CollectSupport()
+					}
+					foundCore = true
```
Comment (mborodin-px) [inline tests/common.go:1836]: you don't need to continue if core file was found. Just `break` from the loop as the `CollectSupport()` method has already collected all needed logs and diags from all nodes

File: tests/common.go [hunk 7]
```diff
@@ -1827,12 +1828,18 @@ func PerformSystemCheck() {
...
 				}
 				log.FailOnError(err, "Error occurred while checking for core on node %s", n.Name)
 			}
+			if foundCore {
+				log.Fatalf("Core generated, please check logs for more details")
+			}
```
Comment (mborodin-px) [inline tests/common.go:1842]: don't need this as above comment

File: tests/common.go (hunk 6 above)
Comment (CharudathGopal) [reply tests/common.go:1836]: I am continuing the loop to logs message if there is core in other nodes too. it wont collect support bundle but just logs the info if core found. Let me know if this doesn't make sense

**Comment by CharudathGopal** [comment]
Thanks for review Maks. Taken care of comments.

File: tests/common.go [hunk 8]
```diff
@@ -1827,11 +1827,11 @@ func PerformSystemCheck() {
 					TimeBeforeRetry: 10 * time.Second,
 				})
 				if len(file) != 0 || err != nil {
-					log.Errorf("an error occurred, collecting bundle")
+					dash.Errorf("Core file was found on node %s, Core Path: %s", n.Name, file)
+					// Collect Support Bundle only once
 					CollectSupport()
-					dash.VerifySafely(file == "", true, fmt.Sprintf("Core generated on node %s, Core Path: %s", n.Name, file))
+					log.Fatalf("Core generated, please check logs for more details")
```
Comment (mborodin-px) [inline tests/common.go:1833]: you have to add `break` here to get out from loop

**Review by mborodin-px — APPROVED** [review]
LGTM

File: tests/common.go (hunk 8 above)
Comment (CharudathGopal) [reply tests/common.go:1833]: no need @mborodin-px , once the log.Fatal is called it will fail the test and return

**Review by kshithijiyer-px — APPROVED** [review]

**Pull request merged**